### PR TITLE
Add required meta service client interfaces

### PR DESCRIPTION
**Pull request by HelloHorizon**
Add required client interface with AlluxioStatusException

### CODE REVIEW

**Comment by HelloHorizon** [comment]
@yuzhu please check this, we need the consistent database, table, ColumnStatisticsObj and partition definitions from client to metastore. I hacked with hive class here.

**Comment by HelloHorizon** [comment]
jenkins, test this please

File: experimental/table/client/pom.xml [hunk 1]
```diff
@@ -67,6 +67,11 @@
       <groupId>io.netty</groupId>
       <artifactId>netty-all</artifactId>
```
Comment (apc999) [inline experimental/table/client/pom.xml]: why netty is required?

File: experimental/table/client/src/main/java/alluxio/client/catalog/CatalogMasterClient.java [hunk 2]
```diff
@@ -47,7 +53,121 @@ public static CatalogMasterClient create(MasterClientContext conf) {
    * Get a list of all database names.
    *
    * @return list of database names
-   * @throws IOException
+   * @throws AlluxioStatusException
+   */
+  List<String> getAllDatabase() throws AlluxioStatusException;
```
Comment (apc999) [inline experimental/table/client/src/main/java/alluxio/client/catalog/CatalogMasterClient.java]: `getAllDatabase` or `getAllDatabases`?

File: experimental/table/client/src/main/java/alluxio/client/catalog/CatalogMasterClient.java [hunk 3]
```diff
@@ -47,7 +53,121 @@ public static CatalogMasterClient create(MasterClientContext conf) {
...
+
+  /**
+   * Get database metadata.
```
Comment (apc999) [inline experimental/table/client/src/main/java/alluxio/client/catalog/CatalogMasterClient.java]: Nit: Gets, similar comment applies to all other method javadoc

File: experimental/table/client/src/main/java/alluxio/client/catalog/CatalogMasterClient.java [hunk 4]
```diff
@@ -12,11 +12,17 @@
 package alluxio.client.catalog;
 
 import alluxio.Client;
+import alluxio.exception.status.AlluxioStatusException;
 import alluxio.master.MasterClientContext;
 
+import org.apache.hadoop.hive.metastore.api.ColumnStatisticsObj;
```
Comment (apc999) [inline experimental/table/client/src/main/java/alluxio/client/catalog/CatalogMasterClient.java]: is importing data structures from Hive Metastore a workaround or final design?

**Comment by HelloHorizon** [comment]
jenkins, test this please

File: experimental/table/client/src/main/java/alluxio/client/catalog/CatalogMasterClient.java (hunk 2 above)
Comment (HelloHorizon) [reply experimental/table/client/src/main/java/alluxio/client/catalog/CatalogMasterClient.java]: it should be getAllDatabases.

File: experimental/table/client/src/main/java/alluxio/client/catalog/CatalogMasterClient.java (hunk 4 above)
Comment (HelloHorizon) [reply experimental/table/client/src/main/java/alluxio/client/catalog/CatalogMasterClient.java]: The importing data structures from Hive metastore is a workaround to simply implementation. We expect we have our data structure in the final design.

**Comment by HelloHorizon** [comment]
jenkins, test this please

**Comment by HelloHorizon** [comment]
jenkins, test this please

File: experimental/table/client/pom.xml [hunk 5]
```diff
@@ -67,6 +67,11 @@
...
     </dependency>
+    <dependency>
+      <groupId>org.apache.hive</groupId>
+      <artifactId>hive-metastore</artifactId>
+      <version>3.0.0</version>
```
Comment (yuzhu) [inline experimental/table/client/pom.xml]: maybe we should make sure that we use a consistent `hive-metastore` dependency in our project? see  server pom. I don't have a preference on which to use, as long as we are consistent

Comment (HelloHorizon) [reply experimental/table/client/pom.xml]: @yuzhu  Since I use version 3.0.0 in the connector side, should we keep 3.0.0 here?

Comment (yuzhu) [reply experimental/table/client/pom.xml]: yeah we should put the dependency into the parent, and use `dependencyManagement` declare it there. and only include it when necessary, in server and client.

**Comment by HelloHorizon** [comment]
jenkins, test this please

File: experimental/table/client/pom.xml (hunk 1 above)
Comment (yuzhu) [reply experimental/table/client/pom.xml]: @HelloHorizon could you remove netty from both the client pom and the server pom, i don't think we need those.

Comment (HelloHorizon) [reply experimental/table/client/pom.xml]: @yuzhu sure.

**Comment by HelloHorizon** [comment]
jenkins, test this please

**Comment by HelloHorizon** [comment]
jenkins, test this please

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Pull request merged**